analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Conversion.cs

```diff
@@ -26,7 +28,22 @@ internal sealed class Conversion : SimpleProcessor<IConversionOperationWrapper>
         operation.ToConversion();
 
     protected override ProgramState Process(SymbolicContext context, IConversionOperationWrapper conversion) =>
-        context.State[conversion.Operand] is { } value && conversion.OperatorMethod is null // Built-in conversions only
-            ? context.State.SetOperationValue(context.Operation, value)
+        context.State[conversion.Operand] is { } value
+        && PropagateSymbolValue(value, conversion) is { } propagatedValue
+            ? context.State.SetOperationValue(context.Operation, propagatedValue)
             : context.State;
+
+    private static SymbolicValue PropagateSymbolValue(SymbolicValue value, IConversionOperationWrapper conversion)
+    {
+        if (conversion.OperatorMethod is not null)
+        {
+            // Don't propagate state in user-defined conversions
+            value = SymbolicValue.Constraintless;
+        }
+        if (conversion.Type.IsNonNullableValueType())
+        {
+            value = value.WithConstraint(ObjectConstraint.NotNull);
+        }
```

Comment:
What about changing `NonNullableValueTypeCheck` from `PreProcess` to `PostProcess` and move it to the end of the check registration list?

If I understand the relation right, the problem is that the `PreProcess` is thrown away. And we want it there to be always. Like absolutely always.

Reply:
Yes. That might work also. I could give this another shot in parallel to this PR or as a follow-up after this PR is merged. What would you prefer?

Reply:
The another PR would need to undo this work. Let's give it a shot in parallel PR to see if it actually works. And merge the other one in case it wins.

Reply:
> I'm not sure what that means

With the `NonNullableValueTypeCheck` we do the same thing: `context.SetOperationConstraint(ObjectConstraint.NotNull);` We discussed whether NonNullableValueTypeCheck should be a "Pre" or "Post" check. I wanted to show that for some operation-kinds we do it right in the middle via NotNullOperation. So running NonNullableValueTypeCheck before or afterward makes sense both IMO. Let's see what feels better in practice.

Reply:
"In middle" is just a nature of the operation processors. The `PreProcess` has some significant drawbacks that we didn't see before, so we need to try the `PostProcess` to see if that fixes the problems.

Reply:
I have created #6922. PostProcess just works. I converted this to a draft until we decide which way to go.